src/rascal/structure_managers/adaptor_increase_maxorder.hh

```diff
@@ -343,36 +364,22 @@ namespace rascal {
 
     using NextOrderLoop =
         AddOrderLoop<Order + 1, NeighbourListType, IsCompactCluster,
-                     (Order + 1 == OldMaxOrder)>;
+                     (Order + 1 <= OldMaxOrder)>;
 
     using AtomClusterRef_t =
         typename ManagerImplementation::template ClusterRef<1>;
 
     // do nothing, if MaxOrder is not reached, except call the next order
-    static void loop(AtomClusterRef_t & atom, ClusterRef_t & cluster,
-                     size_t start_index,
+    static void loop(AtomClusterRef_t & atom,
                      AdaptorMaxOrder<ManagerImplementation> & manager) {
-      size_t new_start_index{
-          (start_index + 1) *
-          static_cast<int>(NeighbourListType ==
-                           AdaptorTraits::NeighbourListType::half)};
-
-      using Iterator_t = typename ClusterRef_t::iterator;
-      using NextCluster_t = typename Iterator_t::value_type;
-      for (Iterator_t next_cluster_it{cluster.get_iterator_at(start_index)};
-           next_cluster_it != cluster.end(); ++next_cluster_it) {
-        auto && next_cluster{*next_cluster_it};
-        constexpr size_t previous_order{NextCluster_t::order() - 1};
-        auto & next_cluster_indices{
-            std::get<previous_order>(manager.cluster_indices_container)};
+      // copy the cluster indices of all orders below
+      for (auto && cluster : atom.template get_clusters_of_order<Order>(0)) {
+        auto & cluster_indices{
+            std::get<Order - 1>(manager.cluster_indices_container)};
         // keep copying underlying cluster indices, they are not changed
-        auto indices{next_cluster.get_cluster_indices()};
-        next_cluster_indices.push_back(indices);
-
-        NextOrderLoop::loop(atom, next_cluster, new_start_index, manager);
-        if (NeighbourListType == AdaptorTraits::NeighbourListType::half) {
-          ++new_start_index;
```

Comment:
Is our discussion from the  beginning of the week not related to this functionality?
So the way to add neighbours is completely different from the master branch now, right?

Reply:
yes it is

Reply:
Which question is answered with this comment? :)

Reply:
both